src/core/lib/transport/metadata_batch.h

```diff
@@ -553,6 +548,17 @@ struct GrpcStatusContext {
 
 namespace metadata_detail {
 
+// Build a key/value formatted debug string.
+class DebugStringBuilder {
+ public:
+  void Add(absl::string_view key, absl::string_view value);
+
+  std::string TakeOutput() { return std::move(out_); }
```

Comment:
Isn't it UB to do anything to a `DebugStringBuilder` object after you've called `TakeOutput`? I think `out_` will be in what's defined as a "valid but unspecified state". How about instead something like `TakeAndClearOutput` that reinitializes `out_` to a known empty state?

Reply:
commented that such usage would be ub.